models/exposure.py

```diff
@@ -366,6 +387,10 @@ def __init__(self, current_file=None, invent_filepath=True, **kwargs):
         # We will run this exact code again later so that the keywords
         # can override what's detected from the header.
         self.set_attributes_from_dict( kwargs )
+        # NOTE: since above reads directly from header, it will often set
+        # a full filter name without an instrument, which must be detected
+        # and fixed below before making anything permanent (filepaths,
+        # provenances etc).
```

Comment:
I wonder if it's worth putting in some explicit code right here to look at the _filter attribute.  If it's not None, and the isntrument isn't None, call the instrument's `get_short_filter_name` method.

Reply:
We'll probably need the same code in Image.

Reply:
(Just make an issue for this, don't write it in now.)